spec/models/role_spec.rb

```diff
@@ -0,0 +1,34 @@
+require 'spec_helper'
+
+#  Copyright (c) 2012-2022, insieme Schweiz. This file is part of
+#  hitobito_insieme and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito_insieme.
+
+describe Role do
+    # we don't want to check the test roles of the base project
+    base_project_testroles = [
+      Group::BottomGroup::Leader,
+      Group::BottomGroup::Member,
+      Group::BottomLayer::Leader,
+      Group::BottomLayer::LocalGuide,
+      Group::BottomLayer::Member,
+      Group::GlobalGroup::Leader,
+      Group::GlobalGroup::Member,
+      Group::TopGroup::Leader,
+      Group::TopGroup::LocalGuide,
+      Group::TopGroup::Secretary,
+      Group::TopGroup::LocalSecretary,
+      Group::TopGroup::Member,
+      Group::TopLayer::TopAdmin,
+    ]
+
+    (Role.subclasses - base_project_testroles).each do |role|
+      it "#{role.name} has permissions so must have two_factor_authentication_enforced=true" do
+        if role.permissions.present?
```

Comment:
Ist hier nicht die Reihenfolge verkehrt und müsste eigentlich so herum sein?

```suggestion
      if role.permissions.present?
        it "#{role.name} has permissions so must have two_factor_authentication_enforced=true" do
```

Übersehe ich da was?

Reply:
Also der code testet schon korrekt, aber es werden unnötig viele tests generiert.
Habe das deinem Vorschlag entsprechend angepasst :+1: